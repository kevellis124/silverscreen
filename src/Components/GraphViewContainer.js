import React, { Component } from 'react';
import {Scatter} from 'react-chartjs-2';
import DataParser from '../DataParsing/dataParsing';
import DatasetAggregator from '../DataParsing/datasetAggregator';
import Api from '../Api/Api';
import {breakingBadData, sampleGraphOptions} from '../SampleData';
import {searchForShow} from '../Api/NetworkAccess'

const parser = new DataParser();
const aggregator = new DatasetAggregator();
class GraphViewContainer extends Component {

    constructor(props){

        super(props);
        this.state = {
            graphTitle: "Movie/Show Title",
            graphData:  {},
            graphOptions: sampleGraphOptions,
            searchTerm: "Breaking Bad",
            loading: false
        };
    }

    handleTitleSearch = () => {
        this.setState({loading:true});
        //move this logic
        Api.fetchShowDataOmdb(this.state.searchTerm)
            .then(json => parser.seriesParse(json))
            .then(parsedSeriesResponse => aggregator.takeSeriesInfoAndReturnEpisodeData(parsedSeriesResponse))
            .then(seasons => {
                return {
                    labels: [],
                    datasets: seasons
                }
            })
            .then(dataSets => aggregator.setXCoordinatesOfData(dataSets))
            .then(graphData => this.setState({
                graphData: graphData,
                loading:false
            }));
    };
5
    render() {
        //set the graph data for development. Will remove later.
        let graphData = breakingBadData;
        let graphOptions = sampleGraphOptions;

        return (
            <div>
                <Scatter data={this.state.graphData} options={this.state.graphOptions}/>
                <h1>{this.state.graphTitle}</h1>
                <textarea value={this.state.searchTerm} onChange={(event) => {this.setState({searchTerm: event.target.value});}}/>
                <button disabled={this.state.loading} onClick={this.handleTitleSearch}>Submit</button>
                <h1 style={{fontSize : '100px'}} >BETA</h1>
            </div>)
    }
};

export default GraphViewContainer;