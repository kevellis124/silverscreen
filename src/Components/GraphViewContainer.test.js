import React from 'react';
import GraphViewContainer from './GraphViewContainer';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

it('renders correctly', () => {
    const tree = toJson(shallow(<GraphViewContainer></GraphViewContainer>));
    expect(tree).toMatchSnapshot();
});

