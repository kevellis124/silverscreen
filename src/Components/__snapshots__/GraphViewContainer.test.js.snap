// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <Scatter
    data={Object {}}
    options={
      Object {
        "scales": Object {
          "yAxes": Array [
            Object {
              "ticks": Object {
                "beginAtZero": true,
                "display": true,
                "suggestedMax": 10,
                "suggestedMin": 3,
              },
            },
          ],
        },
        "title": Object {
          "display": true,
          "text": "The Silver Screen",
        },
        "tooltips": Object {
          "callbacks": Object {
            "label": [Function],
          },
        },
      }
    }
  />
  <h1>
    Movie/Show Title
  </h1>
  <textarea
    onChange={[Function]}
    value="Breaking Bad"
  />
  <button
    disabled={false}
    onClick={[Function]}
  >
    Submit
  </button>
  <h1
    style={
      Object {
        "fontSize": "100px",
      }
    }
  >
    BETA
  </h1>
</div>
`;
